ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1482,8 +1522,11 @@ protected void maybeProcessCallbacks() {
               if (!resolveRange(totalSize)) {
                 routerMetrics.getFirstDataChunkLatencyMs.update(System.currentTimeMillis() - submissionTimeMs);
                 routerMetrics.getDataChunkLatencyMs.update(System.currentTimeMillis() - initializedTimeMs);
-                chunkIndexToBuf.put(0, filterChunkToRange(decryptedBlobContent));
-                numChunksRetrieved.set(1);
+                ByteBuf decompressedContent = decompressContent(decryptedBlobContent);
```

Comment:
Do we need to decompress content before calling `resolveRange(totalSize)` above? Else, it seems like we might try to resolve the byte range against compressed content size which might be incorrect. For example, start offset of range request could be greater than compressed content size.

Reply:
That is correct, content must be decompressed first before applying the range.  ResolveRange() is calculated based on the chunk sizes, not compressed content size, so there should have no impact.  Compression is hidden to the entire put operation except serialize and de-serialize to blob content.